macros/cross_db_utils/xdb_deprecation_warning.sql

```diff
@@ -0,0 +1,4 @@
+{% macro xdb_deprecation_warning(macro, package, model) %}
+    {%- set error_message = "Warning: the `" ~ macro ~"` macro is no longer available in dbt_utils and backwards compatibility will be removed in a future version. Use `dbt." ~ macro ~ "` instead. The " ~ package ~ "." ~ model ~ " model triggered this warning." -%}
+    {%- do exceptions.warn(error_message) -%}
+{% endmacro %}
```

Comment:
@jtcohen6 and I decided [here](https://github.com/dbt-labs/docs.getdbt.com/pull/1619#pullrequestreview-1021406942) to omit the `dbt` namespace for the [public-facing documentation](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros) for cross-database macros.

I feel the public-facing documentation and the deprecation warning should be in tight alignment; it's confusing for end users when they differ.

There are two different _permissible_ options here (with `dbt.` prefix or without). I wonder if [this](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md) style guide or somewhere else would be good to document our explicit recommendation for which option to use?

```suggestion
{% macro xdb_deprecation_warning(macro, package, model) %}
    {%- set error_message = "Warning: the `" ~ macro ~"` macro is no longer available in dbt_utils and backwards compatibility will be removed in a future version. Drop the `dbt_utils.` prefix and use just `" ~ macro ~ "` instead. The " ~ package ~ "." ~ model ~ " model triggered this warning." -%}
    {%- do exceptions.warn(error_message) -%}
{% endmacro %}
```

Reply:
I 💯 agree with your desire for consistency of advice, especially for the exact same functions. 

_Ideally_ that should be true for all macros across the board, which would mean that we should have no `dbt.` to be in alignment with other out of the box macros (e.g it would be weird for someone to start going around saying `{{ dbt.ref('my_model') }}`. 

BUT. 

My gut feeling on why these ones are different is that they get a bit uncanny-valley-y. If I am new to dbt, then coming across `{{ dateadd() }}`, `{{ except() }}` etc could feel quite confusing - _do I need to put all SQL keywords into Jinja braces?_ (In fairness, people shouldn't come across this in their early days, because it should all be deep in the bowels of packages' code.)

If we do `{{ dbt.dateadd() }}`, then it's clear that it's behaviour provided by dbt, even though it looks a bunch like standard SQL.

This sort of matches up with something else I've been thinking of, which is the shorthand node selection syntax being the same for a package or a model or a directory (`dbt run -s something`). I think we're doing a disservice letting the default advice be "type whatever you want! We'll work it out!" versus the more explicit `package:something`, `path:something`. But that's a different issue for a different day. 

If you or jerco feel very strongly in the opposite direction, I'm willing to be talked around. I just like to know where my magic is coming from 🐰🎩

Reply:
Jerco and I both considered the two options and decided on the version that is shorter to read and write and aligns with `{{ ref() }}`, `{{ source() }}`, etc. i.e., without `dbt.` prefix.

Since we both had the same preference, we didn't discuss it in detail.